gcloud-java-examples/pom.xml

```diff
@@ -22,6 +22,12 @@
       <artifactId>gcloud-java</artifactId>
       <version>${project.version}</version>
     </dependency>
+      <dependency>
```

Comment:
I am fine with using far jar for examples. Though for consistency I would like all examples to use the same (so if we do it here, lets also change the other examples in a separate PR or create an issue for it).

I also think we should pick one approach for creating the fat-jar and my preference would be to use shade.

Also, by "is that necessary" I mean as a dependency (also indent is wrong) as oppose to a plugin configured to run in the package stage.